biorbd_optim/interfaces/acados_interface.py

```diff
@@ -201,6 +225,25 @@ def __set_costs(self, ocp):
         else:
             raise RuntimeError("Available acados cost type: 'LINEAR_LS', 'NONLINEAR_LS' and 'EXTERNAL'.")
 
+    def __init_and_update_solver(self):
+        for n in range(self.acados_ocp.dims.N):
+            self.ocp_solver.cost_set(n, "yref", np.concatenate([data[n] for data in self.y_ref])[:, 0])
+            # TODO deal with non Instant.EACH_FRAME initializations
```

Comment:
It is suppose to be automatic by using 
`self.ocp.nlp[0].X_bounds.min.evaluate_at(shooting_point=k)` instead of referencing the actual matrix